calico_kubernetes/calico_kubernetes_cni.py

```diff
@@ -193,32 +220,30 @@ def _container_remove(hostname, orchestrator_id, container_id):
     :param orchestrator_id (str): Specifies orchestrator
     :param container_id (str):
     """
+    # Un-assign the IP address by calling out to the IPAM plugin
+    _unassign_ip_address()
+
     # Find the endpoint ID. We need this to find any ACL rules
     try:
-        endpoint = datastore_client.get_endpoint(hostname=hostname,
+        endpoint = _datastore_client.get_endpoint(hostname=hostname,
                                                  orchestrator_id=orchestrator_id,
                                                  workload_id=container_id)
     except KeyError:
         _log.error("Container %s doesn't contain any endpoints" % container_id)
         sys.exit(1)
 
-    # Remove any IP address assignments that this endpoint has
-    for net in endpoint.ipv4_nets | endpoint.ipv6_nets:
-        assert net.size == 1, "Only 1 address allowed per endpoint. Found in network: %s" % net
-        datastore_client.unassign_address(None, net.ip)
-
     # Remove the endpoint
     remove_veth(endpoint.name)
 
     # Remove the container from the datastore.
-    datastore_client.remove_workload(hostname=hostname,
+    _datastore_client.remove_workload(hostname=hostname,
                                      orchestrator_id=orchestrator_id,
                                      workload_id=container_id)
 
     _log.info("Removed Calico interface from %s" % container_id)
 
 
-def _set_profile_on_endpoint(endpoint, profile_name):
+def _set_profile_on_endpoint(container_id, namespace, endpoint, pod_name, profile_name):
```

Comment:
I dont really like the name of this function since we're doing a lot more than just setting the profile on endpoint. How about `_configure_profile`?